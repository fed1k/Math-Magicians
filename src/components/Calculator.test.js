import React from "react";
import Design from "./Calculator";
import renderer from 'react-test-renderer';

it('should render correctly', ()=>{
    const data = {
        total: '',
        operation: '',
        next: '0'
      }
    const buttonNames = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
    const tree = renderer
        .create( <Design obj={data} data={buttonNames} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})