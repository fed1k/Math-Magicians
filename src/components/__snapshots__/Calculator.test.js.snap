// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="console"
  >
    0
  </div>
  <button
    onClick={[Function]}
  >
    AC
  </button>
  <button
    onClick={[Function]}
  >
    +/-
  </button>
  <button
    onClick={[Function]}
  >
    %
  </button>
  <button
    onClick={[Function]}
  >
    รท
  </button>
  <button
    onClick={[Function]}
  >
    7
  </button>
  <button
    onClick={[Function]}
  >
    8
  </button>
  <button
    onClick={[Function]}
  >
    9
  </button>
  <button
    onClick={[Function]}
  >
    x
  </button>
  <button
    onClick={[Function]}
  >
    4
  </button>
  <button
    onClick={[Function]}
  >
    5
  </button>
  <button
    onClick={[Function]}
  >
    6
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    1
  </button>
  <button
    onClick={[Function]}
  >
    2
  </button>
  <button
    onClick={[Function]}
  >
    3
  </button>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    0
  </button>
  <button
    onClick={[Function]}
  >
    .
  </button>
  <button
    onClick={[Function]}
  >
    =
  </button>
</div>
`;
